File: src/main/scala/chapter01/Student.scala
package chapter01

class Student(var name: String,val age: Int) {

  def printInfo(): Unit = {
    println("Name = " + " " + this.name + " age = " + age + " school = " + Student.school)
  }
}

// 引入伴生对象，解决没有static
object Student {
  val school: String = "ecust"

  def main(args: Array[String]): Unit = {
    new Student("ls",15).printInfo()
    new Student("xs",16).printInfo()
  }
}

File: src/main/java/Student.java
import javax.sound.midi.Soundbank;

public class Student {
    private String Name;
    private Integer age;
    private static String school = "ecust";

    public Student(String name, Integer age) {
        Name = name;
        this.age = age;
    }

    @Override
    public String toString() {
        return "Student{" +
                "Name='" + Name + '\'' +
                ", age=" + age + ", school=" + school +
                '}';
    }

    public static void main(String[] args) {
        Student zs = new Student("zs", 1);
        Student ls = new Student("ls", 2);
        System.out.println(zs.toString());
        System.out.println(ls.toString());
    }
}
